File: backend/ExpenseService/Models/Expense.cs
namespace ExpenseService.Models;

public class Expense
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TravelPlanId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateOnly Date { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/models/Expense.js
export const EXPENSE_CATEGORIES = ['Transport', 'Accommodation', 'Food', 'Tickets', 'Shopping', 'Other'];

export class Expense {
  constructor(data = {}) {
    this.id = data.id || '';
    this.travelPlanId = data.travelPlanId || '';
    this.name = data.name || '';
    this.category = data.category || 'Other';
    this.amount = data.amount || 0;
    this.date = data.date || '';
    this.description = data.description || '';
    this.createdAt = data.createdAt || null;
  }
}
